tools/code_format/check_format.py

```diff
@@ -16,232 +16,120 @@
-INCLUDE_ANGLE = "#include <"
-INCLUDE_ANGLE_LEN = len(INCLUDE_ANGLE)
-X_ENVOY_USED_DIRECTLY_REGEX = re.compile(r'.*\"x-envoy-.*\".*')
-DESIGNATED_INITIALIZER_REGEX = re.compile(r"\{\s*\.\w+\s*\=")
-MANGLED_PROTOBUF_NAME_REGEX = re.compile(r"envoy::[a-z0-9_:]+::[A-Z][a-z]\w*_\w*_[A-Z]{2}")
-HISTOGRAM_SI_SUFFIX_REGEX = re.compile(r"(?<=HISTOGRAM\()[a-zA-Z0-9_]+_(b|kb|mb|ns|us|ms|s)(?=,)")
-TEST_NAME_STARTING_LOWER_CASE_REGEX = re.compile(r"TEST(_.\(.*,\s|\()[a-z].*\)\s\{")
-EXTENSIONS_CODEOWNERS_REGEX = re.compile(r'.*(extensions[^@]*\s+)(@.*)')
-CONTRIB_CODEOWNERS_REGEX = re.compile(r'(/contrib/[^@]*\s+)(@.*)')
-COMMENT_REGEX = re.compile(r"//|\*")
-DURATION_VALUE_REGEX = re.compile(r'\b[Dd]uration\(([0-9.]+)')
-PROTO_VALIDATION_STRING = re.compile(r'\bmin_bytes\b')
-OLD_MOCK_METHOD_REGEX = re.compile("MOCK_METHOD\d")
-# C++17 feature, lacks sufficient support across various libraries / compilers.
-FOR_EACH_N_REGEX = re.compile("for_each_n\(")
-# Check for punctuation in a terminal ref clause, e.g.
-# :ref:`panic mode. <arch_overview_load_balancing_panic_threshold>`
-DOT_MULTI_SPACE_REGEX = re.compile("\\. +")
-FLAG_REGEX = re.compile("RUNTIME_GUARD\((.*)\);")
-
-# yapf: disable
-PROTOBUF_TYPE_ERRORS = {
-    # Well-known types should be referenced from the ProtobufWkt namespace.
-    "Protobuf::Any":                    "ProtobufWkt::Any",
-    "Protobuf::Empty":                  "ProtobufWkt::Empty",
-    "Protobuf::ListValue":              "ProtobufWkt::ListValue",
-    "Protobuf::NULL_VALUE":             "ProtobufWkt::NULL_VALUE",
-    "Protobuf::StringValue":            "ProtobufWkt::StringValue",
-    "Protobuf::Struct":                 "ProtobufWkt::Struct",
-    "Protobuf::Value":                  "ProtobufWkt::Value",
-
-    # Other common mis-namespacing of protobuf types.
-    "ProtobufWkt::Map":                 "Protobuf::Map",
-    "ProtobufWkt::MapPair":             "Protobuf::MapPair",
-    "ProtobufUtil::MessageDifferencer": "Protobuf::util::MessageDifferencer"
-}
-# yapf: enable
-
-LIBCXX_REPLACEMENTS = {
-    "absl::make_unique<": "std::make_unique<",
-}
-
-CODE_CONVENTION_REPLACEMENTS = {
-    # We can't just remove Times(1) everywhere, since .Times(1).WillRepeatedly
-    # is a legitimate pattern. See
-    # https://github.com/google/googletest/blob/master/googlemock/docs/for_dummies.md#cardinalities-how-many-times-will-it-be-called
-    ".Times(1);": ";",
-    # These may miss some cases, due to line breaks, but should reduce the
-    # Times(1) noise.
-    ".Times(1).WillOnce": ".WillOnce",
-    ".Times(1).WillRepeatedly": ".WillOnce",
-    "Stats::ScopePtr": "Stats::ScopeSharedPtr",
-}
-
-UNSORTED_FLAGS = {
-    "envoy.reloadable_features.activate_timers_next_event_loop",
-    "envoy.reloadable_features.grpc_json_transcoder_adhere_to_buffer_limits",
-}
+# The way this script is currently used (ie no bazel) it relies on system deps.
+# As `pyyaml` is present in `envoy-build-ubuntu` it should be safe to use here.
```

Comment:
This is not just a Docker tool, this is part of normal developer flow.

Reply:
at the momemt its run from hooks and ci

it doesnt use bazel and it doesnt have _any_ deps

this change adds a dep - which atm can only be done if it _is_ available in the docker image

Reply:
Are you saying it's run under Docker from the hooks? Or locally?

I'm basically wondering if we need to document this somewhere like https://github.com/envoyproxy/envoy/blob/main/bazel/README.md#quick-start-bazel-build-for-developers or have a venv?

I realize this connects to the overall Bazel story but I don't want to regress the devX.

Reply:
> Are you saying it's run under Docker from the hooks? Or locally?

in CI it uses the docker container (and it seems in some devs workflows)

in the hooks it is called directly

> I'm basically wondering if we need to document this somewhere like https://github.com/envoyproxy/envoy/blob/main/bazel/README.md#quick-start-bazel-build-for-developers

pyyaml needs to be added to the deps i think

> or have a venv?

please lets focus on using bazel - i have removed (almost) all venvs as their use was messy and complex

> I realize this connects to the overall Bazel story 

its a step towards getting this into bazel

> I don't want to regress the devX.

this PR specifically should barely touch the devX (pyyaml becomes a dep of the script, it becomes easier to add in/exclusions etc)

Reply:
i should add that once we do get this into bazel the possibility of running this incrementally opens up which would greatly improve the devX